src/main/scala/com/datastax/gatling/plugin/model/DseCqlAttributes.scala

```diff
@@ -22,33 +24,37 @@ import io.gatling.core.check.Check
-  * @param readTimeout      Read timeout to be used
   * @param idempotent       Set request to be idempotent i.e. whether it can be applied multiple times
-  * @param defaultTimestamp Set default timestamp on request, overriding current system time
+  * @param node             Set the node that should handle this query
+  * @param customPayload    Custom payload for this request
   * @param enableTrace      Whether tracing should be enabled
-  * @param outGoingPayload  Set ByteBuffer custom outgoing Payload
-  * @param serialCl         Serial Consistency Level to be used
-  * @param fetchSize        Set fetchSize of request i.e. paging size
-  * @param retryPolicy      Retry Policy to be used
+  * @param pageSize         Set pageSize (formerly known as fetchSize)
   * @param pagingState      Set paging State wanted
+  * @param routingKey       Sets the key for token-aware routing
+  * @param routingKeyspace  Sets the keyspace for token-aware routing
+  * @param routingToken     Sets the token to use for token-aware routing
+  * @param serialCl         Serial Consistency Level to be used
+  * @param timeout          How long to wait for this request to complete
   * @param cqlStatements    String version of the CQL statement that is sent
   *
   */
-case class DseCqlAttributes(tag: String,
-                            statement: DseStatement[Statement],
-                            cl: Option[ConsistencyLevel] = None,
-                            cqlChecks: List[Check[CqlResponse]] = List.empty,
-                            genericChecks: List[Check[DseResponse]] = List.empty,
-                            userOrRole: Option[String] = None,
-                            readTimeout: Option[Int] = None,
-                            idempotent: Option[Boolean] = None,
-                            defaultTimestamp: Option[Long] = None,
-                            enableTrace: Option[Boolean] = None,
-                            outGoingPayload: Option[Map[String, ByteBuffer]] = None,
-                            serialCl: Option[ConsistencyLevel] = None,
-                            fetchSize: Option[Int] = None,
-                            retryPolicy: Option[RetryPolicy] = None,
-                            pagingState: Option[PagingState] = None,
-                            cqlStatements: Seq[String] = Seq.empty)
+case class DseCqlAttributes[T <: Statement[T], B <: StatementBuilder[B,T]]
+  (tag: String,
+   statement: DseCqlStatement[T, B],
+   cqlChecks: List[DseCqlCheck] = List.empty,
+   cqlStatements: Seq[String] = Seq.empty,
+    /* General attributes */
+   cl: Option[ConsistencyLevel] = None,
+   idempotent: Option[Boolean] = None,
+   node: Option[Node] = None,
+   /* CQL-specific attributes */
+   customPayload: Option[Map[String, ByteBuffer]] = None,
+   enableTrace: Option[Boolean] = None,
+   pageSize: Option[Int] = None,
+   pagingState: Option[ByteBuffer] = None,
+   queryTimestamp: Option[Long] = None,
```

Comment:
Does this parameter replace `defaultTimestamp`?  If so, could you add it in the scaladoc of the case class?

Reply:
Done